src/ingest/TimeRow.h

```diff
@@ -72,6 +70,14 @@ class TimeRow : public nebula::surface::RowData {
     return row_->readLong(field);
   }
 
+  // read from macro combinations if present there
+  std::string_view readString(const std::string& field) const override {
+    if (macroCombinations_.contains(field)) {
+      return macroCombinations_.at(field);
```

Comment:
you definitely want to define these partition columns as dictionaries since they are single-valued columns.
Later we should explore the partition column so that we can completely eliminate the storage for it